### ISSUE TITLE
500 Error when trying to open jupyter notebook via jupyterhub

### ISSUE THREAD

**Issue by phxedmond** [jupyterhub/jupyterhub#2306]
I dont have an issue when opening a notebook via standard jupyter.

I have everything I can think of up to date(including nbconvert).

It was working fine....

Any ideas would be appreciated

Below is the error I get when I try and open a notebook

> [E 2018-11-13 15:56:17.743 SingleUserNotebookApp web:1670] Uncaught exception GET /julia/user/edmond/notebooks/traffic_sources_python/testing.ipynb (10.8.149.109)
> HTTPServerRequest(protocol='http', host='10.107.52.38', method='GET', uri='/julia/user/edmond/notebooks/traffic_sources_python/testing.ipynb', version='HTTP/1.1', remote_ip='10.8.149.109')
> Traceback (most recent call last):
> File "/usr/local/lib/python3.5/dist-packages/tornado/web.py", line 1590, in _execute
> result = method(self.path_args, **self.path_kwargs)
> File "/usr/local/lib/python3.5/dist-packages/tornado/web.py", line 3006, in wrapper
> return method(self, args, **kwargs)
> File "/usr/local/lib/python3.5/dist-packages/notebook/notebook/handlers.py", line 59, in get
> get_custom_frontend_exporters=get_custom_frontend_exporters
> File "/usr/local/lib/python3.5/dist-packages/notebook/base/handlers.py", line 467, in render_template
> return template.render(**ns)
> File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 989, in render
> return self.environment.handle_exception(exc_info, True)
> File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 754, in handle_exception
> reraise(exc_type, exc_value, tb)
> File "/usr/lib/python3/dist-packages/jinja2/_compat.py", line 37, in reraise
> raise value.with_traceback(tb)
> File "/usr/local/lib/python3.5/dist-packages/notebook/templates/notebook.html", line 1, in top-level template code
> {% extends "page.html" %}
> File "", line 2, in top-level template code
> File "/usr/local/lib/python3.5/dist-packages/notebook/templates/page.html", line 154, in top-level template code
> {% block header %}
> File "/usr/local/lib/python3.5/dist-packages/notebook/templates/notebook.html", line 120, in block "header"
> {% for exporter in get_custom_frontend_exporters() %}
> File "/usr/local/lib/python3.5/dist-packages/notebook/notebook/handlers.py", line 19, in get_custom_frontend_exporters
> from nbconvert.exporters.base import get_export_names, get_exporter
> ImportError: No module named 'nbconvert.exporters.base'
> [E 2018-11-13 15:56:17.745 SingleUserNotebookApp log:150] {
> "Cookie": "jupyterhub-user-edmond=2|1:0|10:1542124505|22:jupyterhub-user-edmond|48:ZDFmOGRjYzYtZGMyMi00Mzk5LTk2NmItNDcyNDQxY2U0MTY4|6e3c966fb2620b889939e0dbcc2e7f3b6cb4fcee6133fd75912da86895c78a79; jupyter-hub-token-edmond="2|1:0|10:1541699296|24:jupyter-hub-token-edmond|44:YjFjOGQ2Y2NkOTk5NGNhYzhjMDk4NTE5ZGNmYWUzNzM=|ebcdb02658062d4ab39a985962a0975bbedfc9a3d42c8d69db71b56e89954d99"; _xsrf=2|4464c1f4|074958e7e78ce3490d6b1048d1f754c5|1541702994; username-10-107-52-38-8888="2|1:0|10:1542039453|26:username-10-107-52-38-8888|44:MmQ3ZjllYTQ1OGYyNDk5ODg3Njc5YWFkYWNkOGFhMTQ=|483d02502e07cdedf33d54958c5006d5c15c6e07698d2227a4bdc4426f7bb14a"; jupyterhub-session-id=5067b20a11824cbab169380ce7737762",
> "Cache-Control": "max-age=0",
> "Upgrade-Insecure-Requests": "1",
> "X-Forwarded-For": "10.8.149.109",
> "Connection": "upgrade",
> "Accept-Language": "en-GB,en-US;q=0.9,en;q=0.8",
> "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,/;q=0.8",
> "X-Real-Ip": "10.8.149.109",
> "Host": "10.107.52.38",
> "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36",
> "Referer": "http://10.107.52.38/julia/user/edmond/tree/traffic_sources_python",
> "Accept-Encoding": "gzip, deflate"
> }
> [E 2018-11-13 15:56:17.745 SingleUserNotebookApp log:158] 500 GET /julia/user/edmond/notebooks/traffic_sources_python/testing.ipynb (edmond@10.8.149.109) 4.08ms

**Comment by minrk** [comment]
Can you `python3- m pip install --upgrade nbconvert` and try `python3 -c 'import nbconvert` and `python3 -m pip list`?

**Comment by phxedmond** [comment]
Hey,

Thanks for replying however that did not seem to work.
> Package                                    Version
------------------------------------------ ---------
alabaster                                  0.7.12
alembic                                    1.0.2
asn1crypto                                 0.24.0
async-generator                            1.10
atomicwrites                               1.1.5
attrs                                      18.1.0
awscli                                     1.14.48
Babel                                      2.6.0
backcall                                   0.1.0
bcrypt                                     3.1.4
best                                       1.0
BigQuery-Python                            1.14.0
bleach                                     3.0.2
blinker                                    1.3
boto                                       2.48.0
boto3                                      1.6.8
botocore                                   1.9.8
cachetools                                 2.0.1
causalimpact                               0.1.1
certifi                                    2017.11.5
cffi                                       1.11.5
chardet                                    3.0.4
cloud-init                                 0.7.7
colorama                                   0.3.7
colorlover                                 0.2.1
command-not-found                          0.3
configobj                                  5.0.6
coverage                                   4.4.2
cryptography                               2.2
cufflinks                                  0.14.5
cycler                                     0.10.0
Cython                                     0.28.2
DateTime                                   4.2
decorator                                  4.0.10
defusedxml                                 0.5.0
dill                                       0.2.7.1
docutils                                   0.14
eli5                                       0.8
entrypoints                                0.2.2
fastparquet                                0.1.5
flake8                                     3.5.0
future                                     0.16.0
gapic-google-cloud-datastore-v1            0.15.3
gapic-google-cloud-error-reporting-v1beta1 0.15.3
gapic-google-cloud-logging-v2              0.91.3
google-api-core                            0.1.3
google-api-python-client                   1.6.4
google-auth                                1.2.1
google-auth-oauthlib                       0.2.0
google-cloud                               0.32.0
google-cloud-bigquery                      0.28.0
google-cloud-bigquery-datatransfer         0.1.0
google-cloud-bigtable                      0.28.1
google-cloud-container                     0.1.0
google-cloud-core                          0.28.0
google-cloud-datastore                     1.4.0
google-cloud-dns                           0.28.0
google-cloud-error-reporting               0.28.0
google-cloud-firestore                     0.28.0
google-cloud-language                      1.0.0
google-cloud-logging                       1.4.0
google-cloud-monitoring                    0.28.0
google-cloud-pubsub                        0.30.1
google-cloud-resource-manager              0.28.0
google-cloud-runtimeconfig                 0.28.0
google-cloud-spanner                       0.29.0
google-cloud-speech                        0.30.0
google-cloud-storage                       1.6.0
google-cloud-trace                         0.17.0
google-cloud-translate                     1.3.0
google-cloud-videointelligence             1.0.0
google-cloud-vision                        0.29.0
google-gax                                 0.15.16
google-resumable-media                     0.3.1
googleapis-common-protos                   1.5.3
graphviz                                   0.8.2
grpc-google-iam-v1                         0.11.4
grpcio                                     1.8.3
h5py                                       2.7.1
holidays                                   0.9.3
httplib2                                   0.10.3
idna                                       2.6
imagesize                                  1.1.0
ipykernel                                  5.1.0
ipyparallel                                6.2.3
ipython                                    7.1.1
ipython-genutils                           0.1.0
ipywidgets                                 7.4.2
jedi                                       0.13.1
Jinja2                                     2.8
jmespath                                   0.9.3
joblib                                     0.11
jsonpatch                                  1.10
jsonpointer                                1.9
jsonschema                                 2.6.0
jupyter                                    1.0.0
jupyter-client                             5.2.3
jupyter-console                            6.0.0
jupyter-core                               4.4.0
jupyterhub                                 0.9.4
jupyterlab                                 0.35.4
jupyterlab-server                          0.2.0
language-selector                          0.1
Lifetimes                                  0.9.0.0
llvmlite                                   0.23.2
lxml                                       4.2.5
Mako                                       1.0.7
Markdown                                   2.6.11
MarkupSafe                                 0.23
matplotlib                                 2.1.0
mccabe                                     0.6.1
mistune                                    0.8.4
mmh3                                       2.5.1
mock                                       2.0.0
more-itertools                             4.1.0
nbconvert                                  5.4.0
nbformat                                   4.4.0
nltk                                       3.3
nose                                       1.3.7
notebook                                   5.7.0
numba                                      0.38.1
numpy                                      1.13.3
oauth2client                               3.0.0
oauthlib                                   1.0.3
optimizely-sdk                             2.1.1
packaging                                  18.0
pamela                                     0.2.1
pandas                                     0.22.0
pandas-datareader                          0.7.0
pandas-gbq                                 0.3.0
pandasql                                   0.7.3
pandocfilters                              1.4.2
paramiko                                   2.4.1
parso                                      0.3.1
patsy                                      0.4.1
pbr                                        3.1.1
pem                                        17.1.0
pexpect                                    4.2.1
pickleshare                                0.7.4
Pillow                                     3.1.2
pip                                        10.0.1
pip-conflict-checker                       0.5.0
plotly                                     3.4.0
pluggy                                     0.6.0
ply                                        3.8
prettytable                                0.7.2
prometheus-client                          0.4.2
prompt-toolkit                             2.0.7
proto-google-cloud-datastore-v1            0.90.4
proto-google-cloud-error-reporting-v1beta1 0.15.3
proto-google-cloud-logging-v2              0.91.3
protobuf                                   3.5.1
psutil                                     5.4.3
psycopg2                                   2.7.3.2
ptyprocess                                 0.5.1
py                                         1.5.3
pyarrow                                    0.9.0
pyasn1                                     0.4.2
pyasn1-modules                             0.2.1
pycodestyle                                2.3.1
pycparser                                  2.18
pycurl                                     7.43.0
pydot                                      1.2.4
pyflakes                                   1.6.0
pygame                                     1.9.3
Pygments                                   2.1.3
pygobject                                  3.20.0
PyJWT                                      1.3.0
pymc                                       2.3.6
pymc3                                      3.4.1
PyNaCl                                     1.2.1
pyOpenSSL                                  18.0.0
pyparsing                                  2.2.0
pyserial                                   3.0.1
pysftp                                     0.2.9
pystan                                     2.10.0.0
pytest                                     3.6.0
pytest-flake8                              1.0.1
python-apt                                 1.1.0b1
python-dateutil                            2.6.1
python-debian                              0.1.27
python-editor                              1.0.3
python-oauth2                              1.1.0
python-systemd                             231
pytz                                       2017.2
PyYAML                                     3.12
pyzmq                                      17.1.2
qtconsole                                  4.4.3
requests                                   2.18.4
requests-oauthlib                          0.8.0
retrying                                   1.3.3
roman                                      2.0.0
rsa                                        3.4.2
s3fs                                       0.1.5
s3transfer                                 0.1.13
scikit-learn                               0.19.1
scipy                                      1.0.0
seaborn                                    0.9.0
Send2Trash                                 1.5.0
setuptools                                 38.4.0
simplegeneric                              0.8.1
simpy                                      3.0.11
singledispatch                             3.4.0.3
six                                        1.11.0
sklearn                                    0.0
snowballstemmer                            1.2.1
Sphinx                                     1.8.2
sphinxcontrib-websupport                   1.1.0
SQLAlchemy                                 1.2.13
ssh-import-id                              5.5
statsmodels                                0.8.0
tabulate                                   0.8.2
terminado                                  0.8.1
testpath                                   0.4.2
Theano                                     1.0.1
thinkx                                     1.1.3
thrift                                     0.11.0
tornado                                    5.1.1
tqdm                                       4.23.3
traitlets                                  4.3.2
treeinterpreter                            0.2.1
typing                                     3.6.6
ufw                                        0.35
umap-learn                                 0.3.5
unattended-upgrades                        0.1
uritemplate                                3.0.0
urllib3                                    1.22
wcwidth                                    0.1.7
webencodings                               0.5.1
wheel                                      0.29.0
widgetsnbextension                         3.4.2
wrapt                                      1.10.11
xgboost                                    0.7.post3
xlrd                                       1.1.0
zope.interface                             4.4.3

**Comment by phxedmond** [comment]
any ideas anyone???

**Comment by minrk** [comment]
You were able to import nbconvert outside the notebook in the same env? I'd try removing nbconvert and reinstalling it, since the error suggests that you have only part of nbconvert installed. This would be caused by a bug in the installer most likely (e.g. pip or conda).
